akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -218,28 +196,11 @@ private[remote] class Decoder(
       }
 
       private def resolveRecipient(path: String): OptionVal[InternalActorRef] = {
-        recipientCache.get(path) match {
-          case null ⇒
-            def addToCache(resolved: InternalActorRef): Unit = {
-              // FIXME we might need an efficient LRU cache, or replaced by compression table
-              if (recipientCache.size() >= 1000)
-                recipientCache.clear()
-              recipientCache.put(path, resolved)
-            }
-
-            resolveActorRefWithLocalAddress(path) match {
-              case empty: EmptyLocalActorRef ⇒
-                val pathElements = empty.path.elements
-                if (pathElements.nonEmpty && pathElements.head == "remote")
-                  OptionVal.None
-                else {
-                  addToCache(empty)
-                  OptionVal(empty)
-                }
-              case ref ⇒
-                addToCache(ref)
-                OptionVal(ref)
-            }
+        resolveActorRefWithLocalAddress(path) match {
+          case empty: EmptyLocalActorRef ⇒
+            val pathElements = empty.path.elements
+            if (pathElements.nonEmpty && pathElements.head == "remote") OptionVal.None
```

Comment:
This is for the remote deployment corner case, we need to do this only once, but for this PR it can be like this. I can fix that when it has been merged. Please add a FIXME.